modules/platforms/dotnet/Apache.Ignite.Core/Services/ServiceConfiguration.cs

```diff
@@ -91,6 +100,11 @@ internal void Write(IBinaryRawWriter w)
                 w.WriteObject(NodeFilter);
             else
                 w.WriteObject<object>(null);
+            
+            if (Interceptors != null)
+                w.WriteCollection(Interceptors as ICollection);
```

Comment:
This cast can return null when, because `ICollection<T>` does not inherit from `ICollection`.
Let's add a generic overload for `WriteCollection` to avoid this cast.